### PR TITLE
Add features from set 1 to the page and calls

### PR DESCRIPTION
**Pull request by triazo**
Mostly a request for comments at this point.

### CODE REVIEW

**Comment by lukechampine** [comment]
you can use [`url.Values`](http://golang.org/pkg/net/url/#Values) to encode a query string. Should clean up your API code a bit.

File: api/api.go [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+// As this modules interacts heavily with the Sia API, which sends
+// structs over json, this package uses the simply imports the module
+// packgae where those structs are defined in the core sia package
+
+type ApiLink struct {
+	url string
+
+	// Port is a string as both http requests and command line arguments are strings
+	port string
+}
```
Comment (DavidVorick) [inline api/api.go]: This seems like a strange struct, why do you need it?

Comment (DavidVorick) [inline api/api.go]: also I believe this type already exists in the network package, a 'NetAddress' or something like that.

File: api/api.go [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+		}
+	}
+
+	return link.Get(apiCall)
+}
+
+
+// ExplorerState queries the locally running statu
+func (link *ApiLink) ExplorerState() (explorerStatus modules.ExplorerStatus, err error) {
+	stateJSON, err := link.Get("/blockexplorer/status")
```
Comment (DavidVorick) [inline api/api.go]: Maybe I'm wrong, but it seems like an excessive amount of abstraction

**Comment by DavidVorick** [comment]
I think that, instead of having an ApiLink, you should probably have a different type as your master type. The master type can include an ApiLink, except instead of explicitly declaring the api link, it should just have the 'ip address' and 'host' fields integrated into it. No need to explicitly declare it.

File: hashrate.go [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+func rateString(rate uint64) string {
+	r := float64(rate)
+
+	var i int = 0
+	for r > 1000 {
+		r = r / 1000
+		i++
+	}
+
+	return fmt.Sprintf("%f%s", r, postfixes[i])
```
Comment (DavidVorick) [inline hashrate.go]: need to add a space between the values

File: hashrate.go [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+		i++
+	}
+
+	return fmt.Sprintf("%f %s", r, postfixes[i])
+}
+
+// The hashrate function takes in a slice of block data, each with a
+// timestamp and a target, and returns an estimated number of hashes
+// that were done per second
+func hashrate(blocks []modules.ExplorerBlockData) (rate uint64) {
```
Comment (DavidVorick) [inline hashrate.go]: don't call these blocks, they are not blocks

**Pull request merged**